### PR TITLE
Prow sub: support postsubmit jobs

### PR DESCRIPTION
**Pull request by chaodaiG**

### CODE REVIEW

**Comment by chaodaiG** [comment]
/hold
Wait until #22876 is merged

File: prow/pubsub/subscriber/subscriber.go
Comment (cjwagner) [inline prow/pubsub/subscriber/subscriber.go:239]: I think we're missing this validation for presubmits, please add it there as well.

I think we should require `BaseRef` as well for both presubmits and postsubmits. IIRC it is expected to be populated.

File: prow/pubsub/subscriber/subscriber.go
Comment (cjwagner) [inline prow/pubsub/subscriber/subscriber.go:250]: Reminder to update this to avoid using a nil GitClient as in https://github.com/kubernetes/test-infra/pull/22876

File: prow/pubsub/subscriber/subscriber.go
Comment (chaodaiG) [reply prow/pubsub/subscriber/subscriber.go:242]: done

File: prow/pubsub/subscriber/subscriber.go
Comment (chaodaiG) [reply prow/pubsub/subscriber/subscriber.go]: done

File: prow/pubsub/subscriber/subscriber.go
Comment (cjwagner) [reply prow/pubsub/subscriber/subscriber.go:242]: Still missing the `BaseRef` validation.

Comment (chaodaiG) [reply prow/pubsub/subscriber/subscriber.go:245]: my bad, done

**Review by cjwagner — APPROVED** [review]

**Pull request merged**
